File: leetcode/algorithms/tree/binary-tree-tilt.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function getTiltParams(root: TreeNode | null): [number,number] {
    if(root===null){
        return [0,0];
    }
    
    const leftParams = getTiltParams(root.left);
    const rightParams = getTiltParams(root.right);
    
    return [root.val+leftParams[0]+rightParams[0],Math.abs(leftParams[0]-rightParams[0])+leftParams[1]+rightParams[1]]
}

function findTilt(root: TreeNode | null): number {
    return getTiltParams(root)[1];

};

File: leetcode/algorithms/tree/binary-tree-tilt.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def getTiltProperties(self,root):
        if root==None:
            return [0,0];
        
        leftTiltProperties = self.getTiltProperties(root.left);
        rightTiltProperties = self.getTiltProperties(root.right);
        
        return [root.val+leftTiltProperties[0]+rightTiltProperties[0],abs(leftTiltProperties[0]-rightTiltProperties[0])+leftTiltProperties[1]+rightTiltProperties[1]];
        
    def findTilt(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.getTiltProperties(root)[1]